src/company_gui.cpp

```diff
@@ -81,6 +81,31 @@ static const std::initializer_list<ExpensesType> _expenses_list_capital_costs =
 	EXPENSES_OTHER,
 };
 
+/** Company Ifrastructure Window infrastructure cache type. */
+enum CIWInfrastructureType : int {
+	CIW_ENTRY_HEADER,               ///< Section Header.
+	CIW_ENTRY_BLANK_LINE,           ///< Blank line.
+	CIW_ENTRY_LABEL_ONLY,           ///< Label only entry, no values.
+	CIW_ENTRY_LABEL_VALUES,         ///< Infrastructure name + values (count/cost).
+	CIW_ENTRY_TOTAL_COST,           ///< Infrastructure maintainance total cost.
+};
+
+/**
+ * Infrastructure count and maintenance cost cache container.
+ * container for total infrastructure running cost, monthly maintenance cost and infrastructure count
+ * for drawing the CompanyInfrastructureWindow with a scrollbar.
+*/
+struct CIWInfrastructureCache {
+	CIWInfrastructureType   type;         ///< Section Header.
+	StringID                label;
+	uint                    count;
+	Money                   monthly_cost;
+
+	CIWInfrastructureCache(CIWInfrastructureType type, StringID label, uint count = 0, Money monthly_cost = 0) : type(type), label(label), count(count), monthly_cost(monthly_cost)
+	{
+	}
+};
+
```

Comment:
These definitions should probably be closer to the `CompanyInfrastructureWindow`. Probably even defined inside it, so that the `CIW` prefix is not necessary.